### PR TITLE
[WIP] Write JOSS paper

### PR DESCRIPTION
**Pull request by oxinabox**
@malmaud  I have only really got as far as some references to other tools,
and the pro-forma stuff.
Is there anyone else who should be an author?

Do you want to write some stuff into this PR, and we'll hash it all out together?

Guidelines are here
https://joss.readthedocs.io/en/latest/submitting.html

### CODE REVIEW

**Comment by oxinabox** [comment]
@malmaud   ok, I  have written basically the core of what I think should be in this.
I don't think it is excellently written, but it is there.
Can you do a (heavy) editting pass and add anything missing?

**Comment by malmaud** [comment]
Yes, will do that today.

**Comment by malmaud** [comment]
OK, I'm happy with the version I pushed. Want to take another look? If everything looks good, we can submit.

File: joss_paper/paper.md [hunk 1]
```diff
@@ -24,101 +24,84 @@ bibliography: paper.bib
...
 
+TensorFlow.jl has an elegant, idiomatic Julia syntax. It allows all the usual infix operators such as `+`, `-`, `*` etc. It works seamlessly with Julia's broadcast syntax as well, such as the `.*` operator. This `*` can correspond to matrix multiplication while `.*` corresponds to element-wise multiplication, while Python clients need an explicit `matmul` function. It also allows Julia-style indexing (e.g. `x[:, ii + end÷2]`), and concatenation (e.g. `[A B]`, `[x; y; 1]`). Its goal is to be idiomatic for Julia users, while still preserving all the power and maturity of the TensorFlow system. For example, it allows Julia code to operate on TPUs by virtue of using the same TensorFlow graph syntax as Python's TensorFlow client, even though there is no native Julia TPU compiler.
```
Comment (oxinabox) [inline joss_paper/paper.md]: I just looked this up in the source for python Tensorflow

 - `matmul` is called by the use of infix `@`, which corresponds to our `*`.
 - and `multiply` is called by the use of `*` which corresponds to our `.*`

Can you tweak this?

Comment (oxinabox) [reply joss_paper/paper.md]: Fixed it

**Comment by oxinabox** [comment]
Ok,
I am done.
I am happy for this to be merged then submitted.

If you want to merge it, I can submit it (though it isn't hard).

**Comment by oxinabox** [comment]
@malmaud 
as discussed, I would like to include the version that we submit to JOSS as an appendix to my thesis.

For me to be allowed to do that,
I need your permission to include it,
and agreement that my contributions are correctly summarized as:

> Co-maintainer and second highest contributor to the software. Wrote publication.

If you agree, can you send me an image of your signature (ideally a SVG or PNG or something).
If you are not happy with the idea of just sticking it somewhere public like github,
can you email it to me at lyndon.white@research.uwa.edu.au ?

**Comment by malmaud** [comment]
OK, I submitted it to JOSS. Sure, I'll email you a signature.

**Comment by oxinabox** [comment]
Thanks, got it.

**Pull request merged**